src/Facebook/Facebook.php

```diff
@@ -586,4 +589,70 @@ public function videoToUpload($pathToFile)
     {
         return new FacebookVideo($pathToFile);
     }
+
+    /**
+     * Upload video
+     *
+     * @param $target
+     * @param $pathToFile
+     * @param $accessToken
+     * @param int $maxTransferTries
+     *
+     * @return array
+     *
+     * @throws FacebookUploadException
+     */
+    public function uploadVideo($target, $pathToFile, $accessToken, $maxTransferTries = 5)
+    {
+        $uploader = $this->getResumableUploader($this->app, $accessToken, $maxTransferTries);
+        $endpoint = '/'.$target.'/videos';
+        $chunk = $uploader->start($endpoint, $pathToFile);
+
+        do {
+            $chunk = $uploader->transfer($endpoint, $chunk);
+        } while(!$chunk->isLastChunk());
+
+        return [
+            'videoId' => $chunk->getVideoId(),
+            'success' => $uploader->finish($endpoint, $chunk->getUploadSessionId()),
+        ];
+    }
+
+    /**
+     * Instantiates a new video uploader
+     *
+     * @param string $accessToken
+     * @param int $maxTransferTries
+     *
+     * @return FacebookResumableUploader
+     */
+    protected function getResumableUploader($accessToken, $maxTransferTries = 5)
+    {
+        return new FacebookResumableUploader($accessToken, $maxTransferTries);
+    }
+
+    /**
+     * @param FacebookUploadTransferException $exception
+     *
+     * @return array
+     *
+     * @throws FacebookUploadException
+     */
+    public function resumeUpload(FacebookUploadTransferException $exception)
+    {
+        $resumeContext = $exception->getResumeContext();
+
+        $uploader = $this->getResumableUploader($this->app, $resumeContext->accessToken, $resumeContext->maxTransferTries);
```

Comment:
Yes. but in the facebook-php-sdk-v4/src/Facebook/FacebookRequest.php line 197
return $accessTokenEntity->getAppSecretProof($this->app->getSecret());
it requires an app.